compiler/llvm/clangUtil.cpp

```diff
@@ -3830,9 +3831,48 @@ void addDumpIrPass(const PassManagerBuilder &Builder,
   PM.add(createDumpIrPass(llvmPrintIrStageNum));
 }
 
-static void linkLibDevice() {
-  // We follow the directions in https://llvm.org/docs/NVPTXUsage.html#libdevice
+static void linkBitCodeFile(const char *bitCodeFilePath) {
+  GenInfo* info = gGenInfo;
+
+  // load into new module
+  llvm::SMDiagnostic err;
+  auto libdevice = llvm::parseIRFile(bitCodeFilePath, err,
+                                     info->llvmContext);
+  
+  // adjust it
+  const llvm::Triple &Triple = info->clangInfo->Clang->getTarget().getTriple();
+  libdevice->setTargetTriple(Triple.getTriple());
+  libdevice->setDataLayout(info->clangInfo->asmTargetLayoutStr);
+
+  // link
+  llvm::Linker::linkModules(*info->module, std::move(libdevice),
+                            llvm::Linker::Flags::LinkOnlyNeeded);
+}
+
+static std::string determineOclcVersionLib(std::string libPath) {
```

Comment:
I don't know what "oclc" stands for. Is it something too obvious, that I'll learn soon after reading more about AMD, or do we need a short comment here?

Reply:
Apparantly "OpenCL Compiler". Which I find a little weird but if I fail to include the file I'll get an error "lld: error: undefined symbol: __oclc_ISA_version".

I'm guessing AMD uses the same implementation of these math functions whether they're compiled from HIP or OpenCL so some of these bitcode files are being considered as part of the architecture-specific OpenCL implementation.

Reply:
Yeah that makes sense. oclc is a dependency for ocml it looks like: https://github.com/RadeonOpenCompute/ROCm-Device-Libs